Sources/ArgumentParser/Parsing/CommandParser.swift

```diff
@@ -78,6 +78,13 @@ extension CommandParser {
     
     // We should have used up all arguments at this point:
     guard split.isEmpty else {
+      // Check if one of the arguments is an unknown option
```

Comment:
This feels like a weird spot to essentially redo the work done here https://github.com/apple/swift-argument-parser/blob/339baea0a5d8c9b26c6e9be47df8e713f6e26234/Sources/ArgumentParser/Parsing/ArgumentSet.swift#L436. However that error is swallowed here https://github.com/apple/swift-argument-parser/blob/e17dac8424d0275d082dacf00524f2c372d50bf8/Sources/ArgumentParser/Parsing/ArgumentSet.swift#L373-L375 to support nested command structures (https://github.com/apple/swift-argument-parser/issues/23#issuecomment-592983185).

Perhaps there is a better solution, but this is what I came up with 😄

Reply:
`ArgumentSet.first(matching:at:)` could just return an `ArgumentDefinition?` instead of throwing, since we've postponed the actual throwing until this later point.

Reply:
I'd have to look into it. I'm actually still looking at this code as part of #23 so perhaps I can tacke lit there 😅